test/test_testing.py

```diff
@@ -4936,9 +5086,9 @@ def test_register_match_all_prefix(self):
     def test_register_with_result(self):
         self.harness.handle_exec(self.container, ["foo"], result=10)
 
-        with self.assertRaises(pebble.ExecError) as exc:
+        with self.assertRaises(pebble.ExecError) as exc:  # type: ignore
```

Comment:
Why are these two `type: ignore`s necessary? I wouldn't expect them to be needed here, as this is correctly-typed API usage.

Reply:
Ah my bad, I meant to go back to this one. I initially thought it was going to be tricky to cleanly solve, but all it needed was to provide the exact type in the exception. Thanks!

Reply:
Actually, I thought I had figured this out, but I made the type checking happy and the test sad :(

The difference here is that `ExecError` is a Generic, so we need to show that `exc` is a `ExecError[str]`. The type comes from the context manager, which uses the type provided in the `assertRaises` call, but I'm not sure how to make that a specific type. I'll noodle on this a bit more.

Reply:
Hmm, actually pyright is ok with it as-is, it's just Visual Studio that is unhappy. So I guess it's fine then?